invisible_cities/reco/xy_algorithm_test.py

```diff
@@ -217,44 +327,112 @@ def test_is_masked():
-    masked_pos = np.array([(2, 1)])
+    datasipm   = datasipm_3x5
+    x          = np.nan
+    qs         = np.array([[1, 1, 1],
+                           [1, 5, x],
+                           [1, 1, 1],
+                           [1, 6, 1],
+                           [1, 1, 1]]).flatten()
+    pos        = np.stack((datasipm.X.values, datasipm.Y.values), axis=1)
+    ok         = ~np.isnan(qs)
 
     # Corona should return 1 cluster if the masked sipm is taken into account...
     expected_nclusters = 1
-    found_clusters = corona(pos, qs,
-                            msipm         = 6              ,
-                            Qlm           = 4   * units.pes,
-                            new_lm_radius = 1.5 * units.mm ,
-                            pitch         = 1   * units.mm )
+    found_clusters = corona(pos[ok], qs[ok], datasipm,
+                            msipm           = 6              ,
+                            Qthr            = 0   * units.pes,
+                            Qlm             = 4   * units.pes,
+                            new_lm_radius   = 1.5 * units.mm ,
+                            consider_masked = False)
 
     assert len(found_clusters) == expected_nclusters
 
     # ... and two when ignored.
     expected_nclusters = 2
-    found_clusters = corona(pos, qs,
-                            msipm          = 6              ,
-                            Qlm            = 4   * units.pes,
-                            new_lm_radius  = 1.5 * units.mm ,
-                            pitch          = 1   * units.mm ,
-                            masked_sipm    = masked_pos     )
+    found_clusters = corona(pos[ok], qs[ok], datasipm,
+                            msipm           = 6              ,
+                            Qthr            = 0   * units.pes,
+                            Qlm             = 4   * units.pes,
+                            new_lm_radius   = 1.5 * units.mm ,
+                            consider_masked = True)
 
     assert len(found_clusters) == expected_nclusters
+
+
+def test_corona_finds_masked_sipms_correclty(datasipm5x5):
+    # With the appropiate parameters, this should yield a single cluster
+    # with only 25 - 8 = 17 SiPM.
+    datasipm = datasipm5x5
+
+    # For visualization purposes we use o = np.nan and then remove
+    # those values
+    x       = np.nan
+    all_xys = np.stack([datasipm.X.values, datasipm.Y.values], axis=1)
+    all_qs  = np.array([[1, 2, 3, 4, 9], # create slight assymetry
+                        [2, x, x, x, 4], # so the barycenter doesn't
+                        [3, x, 9, x, 3], # fall exactly at the center
+                        [4, x, x, x, 2], # of the array
+                        [5, 4, 3, 2, 1]], dtype=np.float).flatten()
+
+    ok = ~np.isnan(all_qs)
+
+    c = corona(all_xys[ok], all_qs[ok], datasipm,
+               Qthr            =      0,
+               Qlm             =      6,
+                   lm_radius   =      0,
```

Comment:
```suggestion
               lm_radius       =      0,
```

Reply:
I liked how it aligned with `new_lm_radius`, but I don't have a strong opinion

Reply:
Ah, I thought it was a mistake, if you had a purpose there, just leave it as you like it :-).